src/Compilers/CSharp/Portable/Binder/Binder.ValueChecks.cs

```diff
@@ -3587,6 +3614,30 @@ internal bool CheckRefEscape(SyntaxNode node, BoundExpression expr, uint escapeF
                     {
                         return CheckRefEscape(node, conversion.Operand, escapeFrom, escapeTo, checkingReceiver, diagnostics);
                     }
+
+                    Debug.Assert(!conversion.Conversion.IsUserDefined ||
+                        conversion.Conversion.Method.HasUnsupportedMetadata ||
+                        conversion.Conversion.Method.RefKind == RefKind.None);
+                    break;
+
+                case BoundKind.UnaryOperator:
+                    Debug.Assert(expr is BoundUnaryOperator unaryOperator &&
+                        (unaryOperator.MethodOpt is not { } unaryMethod ||
+                        unaryMethod.HasUnsupportedMetadata ||
+                        unaryMethod.RefKind == RefKind.None));
+                    break;
+
+                case BoundKind.BinaryOperator:
```

Comment:
I am wondering if there is also a gap for something like `a += b` where `+` is a user defined operator which may return `b`. [SharpLab](https://sharplab.io/#v2:EYLgZgpghgLgrgJwgZwLQAUoKgW2QYQHsAbYiAYxgEtCA7ZAGhhCmRgYBMQBqAHwAEATAEYAsACh+AZgAESMDLYI4lGfhkBvCTJ0zpe4QDY1MwgAcI2GIQQzuACnVkw7EwioBzABYwAlDIBeAD45Tx8Abm1dfXUAWUcZcgZFcnMIDhNyYX8tcV18xLsAxOFIvIKdfgB2RLL8gF8JeqA=).